specs/sharding/beacon-chain.md

```diff
@@ -184,6 +213,9 @@ class BeaconState(merge.BeaconState):  # [extends The Merge state]
     previous_epoch_attestations: List[PendingAttestation, MAX_ATTESTATIONS * SLOTS_PER_EPOCH]
     current_epoch_attestations: List[PendingAttestation, MAX_ATTESTATIONS * SLOTS_PER_EPOCH]
     # [New fields]
+    # Builder registry.
+    builders: List[Builder, BUILDER_REGISTRY_LIMIT]
+    builder_balances: List[Gwei, BUILDER_REGISTRY_LIMIT]
```

Comment:
Did I understand correctly that these fields are kind of temporary placeholders for now?
May be for simplicity of implementation and testing it makes sense to assign the `builder` role to validators only? 
I.e. only a `validator` _may_ become a `builder`. And thus 'publish fees' are paid between existing validators 
This is also a temporary draft solution of course

Reply:
A registry of builders is closer to my long-term idea than reuse of validators. It may be nice for a prototype, but I don't want to get stuck in a model with validators only. Both for privacy of validators, simplicity to become a builder, and separation of concerns w.r.t. balances, slashings and fees.